### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by cyrusdiego**
- Using Android Navigation Component [link](https://developer.android.com/guide/navigation) to handle navigation within the app. 
- The use of the bottom navigation bar requires the use of fragments for each page rather than an activity. This allows for a seamless transition between pages and allows for use of the Navigation component. 
- This will change the software architecture, specifically the ViewModel and associated LiveData for each page. 

# Experiment: Activities + Navigation Bar
- Notice the screen create a new activity when clicking "Library", we wouldn't want this behavior as it would be a bit weird UX-wise and could eventually lead to perf issues. 
- **[Is it not recommended to use activities with bottom navigation?](https://stackoverflow.com/questions/50119455/is-it-not-recommended-to-use-activities-with-bottom-navigation)**
Screen Recording 2020-10-18 at 3 44 13 PM

### CODE REVIEW

**Comment by cyrusdiego** [comment]
# PR Notes:
- This PR updates our software architecture to accommodate for the navigation component, effectively transitioning into a single activity - multi fragment architecture 
- I added the four main fragments and have left them bare-bones (just setup initial data binding in the source file and layout file) 
- I removed the previously defined Activity files and set the proceeding Activity after Auth to be `MainActivity` 

## Todo
- UI Tests (will need to discuss what is the best solution, Navigation component uses Espresso [link](https://developer.android.com/guide/navigation/navigation-testing)
- Update UML class diagram to reflect changes in software architecture 
- Styling for Navigation Bar 

## Current State: 
 
Screen Recording 2020-10-22 at 7 32 44 PM

File: app/src/main/java/com/example/unlibrary/exchange/ExchangeFragment.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Exchange Fragment
```
Comment (calebschoepp) [inline app/src/main/java/com/example/unlibrary/exchange/ExchangeFragment.java:2]: Nit: No space between

```suggestion
 * ExchangeFragment
```

File: app/src/main/java/com/example/unlibrary/library/LibraryFragment.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Library Fragment
```
Comment (calebschoepp) [inline app/src/main/java/com/example/unlibrary/library/LibraryFragment.java:2]:
```suggestion
 * LibraryFragment
```

File: app/src/main/java/com/example/unlibrary/unlibrary/UnlibraryFragment.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Unlibrary Fragment
```
Comment (calebschoepp) [inline app/src/main/java/com/example/unlibrary/unlibrary/UnlibraryFragment.java:2]:
```suggestion
 * UnlibraryFragment
```

**Review by calebschoepp — CHANGES_REQUESTED** [review]
Couple of nits but all around great work @cyrusdiego . Looking forward to shipping this cause it's gonna make the app real nice.

Just to confirm, each primary fragment will hold it's own ViewModel right? So `LibraryFragment` would instantiate a `Library` ViewModel which would be used by any fragments that `LibraryFragment spawns?

File: app/src/main/java/com/example/unlibrary/unlibrary/UnlibraryFragment.java (hunk 3 above)
Comment (cyrusdiego) [reply app/src/main/java/com/example/unlibrary/unlibrary/UnlibraryFragment.java:2]: Good catch! 

> Just to confirm, each primary fragment will hold its own ViewModel right? So LibraryFragment would instantiate a Library ViewModel which would be used by any fragments that `LibraryFragment spawns?

Yes, that is what I am thinking and @armiantos did some more research in favor of that architecture as well.

**Review by armiantos — COMMENTED** [review]
Looks good to me! Should be good to merge after Caleb's comments are resolved.

## TODO
- Update individual activities to show what they're supposed to show

**Review by calebschoepp — APPROVED** [review]
Looks like you ran the formatter. LGTM!

File: app/src/main/res/layout/fragment_library_book.xml [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
-   <data>
-       <variable name="book" type="com.example.unlibrary.models.Book"/>
-   </data>
+
+    <data>
+
+        <variable
+            name="book"
+            type="com.example.unlibrary.models.Book" />
+    </data>
```
Comment (armiantos) [inline app/src/main/res/layout/fragment_library_book.xml:10]: Was this the auto formatter's work?

**Review by armiantos — COMMENTED** [review]
Last comment before merging, everything else looks good!

File: app/src/main/res/layout/fragment_library_book.xml (hunk 4 above)
Comment (calebschoepp) [reply app/src/main/res/layout/fragment_library_book.xml:10]: Yeah it is

The formatter did that to me too

Comment (armiantos) [reply app/src/main/res/layout/fragment_library_book.xml:10]: Sweet, approved from me :+1:

**Review by armiantos — APPROVED** [review]

**Pull request merged**